src/components/solr-faceted-search.js

```diff
@@ -31,54 +31,60 @@ class SolrFacetedSearch extends React.Component {
     const CsvExportComponent = customComponents.results.csvExport;
     const CurrentQueryComponent = customComponents.searchFields.currentQuery;
     const SortComponent = customComponents.sortFields.menu;
-    const resultPending = results.pending ? (<ResultPendingComponent bootstrapCss={bootstrapCss}/>) : null;
+    const resultPending = results.pending ? (<ResultPendingComponent bootstrapCss={bootstrapCss} />) : null;
 
     const pagination = query.pageStrategy === "paginate" ?
-      <PaginateComponent {...this.props} bootstrapCss={bootstrapCss} onChange={onPageChange}/> :
+      <PaginateComponent {...this.props} bootstrapCss={bootstrapCss} onChange={onPageChange} /> :
       null;
 
     const preloadListItem = query.pageStrategy === "cursor" && results.docs.length < results.numFound ?
       <PreloadComponent {...this.props} /> : null;
 
     return (
-      <div className={cx("solr-faceted-search", {"container": bootstrapCss, "col-md-12": bootstrapCss})}>
-        <SearchFieldContainerComponent bootstrapCss={bootstrapCss} onNewSearch={this.props.onNewSearch}>
+      <div className={cx("solr-faceted-search", { "container": bootstrapCss, "col-md-12": bootstrapCss })}>
+        <SearchFieldContainerComponent
+          bootstrapCss={bootstrapCss}
+          onNewSearch={this.props.onNewSearch}
+          onSetFields={this.props.onSetFields}
+          searchFields={searchFields}>
```

Comment:
In order to create the dropdown menu, we need to input the `searchFields` and new function into the container component.